ddtrace/appsec/internal/protections.py

```diff
@@ -0,0 +1,13 @@
+from typing import Any
+from typing import Mapping
+from typing import TYPE_CHECKING
+
+
+if TYPE_CHECKING:
+    from ddtrace import Span
+
+
+class BaseProtection(object):
```

Comment:
we could make this an `abc.ABC` and mark `process` as an `@abstractmethod`.

I don't think I have a strong opinion on it.